### PR TITLE
Update and add settings for the linting test.

### PR DESCRIPTION
**Pull request by rexwangcc**
https://broadinstitute.atlassian.net/jira/software/projects/GH/boards/530

## Changes:
- [x] Add [Black](https://black.readthedocs.io/en/stable/index.html) as the code formatter for Pipeline-tools.
- [x] Add Flake-8 and its configuration to the repo.
- [x] Update the Travis CI config so we have a linting test before all of the other tests.
- [x] Add `pre-commit` and update the requirements as well as the readme.
- [x] Add some status badges.
 

## Review+QA Instructions:

- **Please expect the test to fail until we merge the [PR 134](https://github.com/HumanCellAtlas/pipeline-tools/pull/134) into this feature branch!!**
- Please check the new multi-stage test on Travis, especially verify that there is a new "linting test" before all tests.
- Please follow the updated readme, make sure the `pre-commit` is installed on your local environment and if feasible, try to make a commit (e.g. add an empty line to the readme, but don't push it), see if the hook works. **Using a virtualenv is highly recommended!**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+repos:
+-   repo: https://github.com/ambv/black
+    rev: 19.3b0
+    hooks:
+    - id: black
+      language_version: python3.6
+      # Using args here is not recommended by Black:
```
Comment (tbl3rd) [inline .pre-commit-config.yaml]: Trailing whitespace here an on following lines.

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,26 @@
...
-install: pip install -U setuptools && pip install -r requirements.txt -r test-requirements.txt
-env:
-script: pytest -v
+install: "pip install -U setuptools && pip install -r requirements.txt -r test-requirements.txt"
+
+jobs:
+  include:
+    # The linting test is fast and cheap so run it first
+    - stage: Linting Test
+      script:
```
Comment (tbl3rd) [inline .travis.yml]: Trailing whitespace here.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (rexwangcc) [reply .pre-commit-config.yaml]: fixed

File: .travis.yml (hunk 2 above)
Comment (rexwangcc) [reply .travis.yml]: fixed

**Pull request merged**